### PR TITLE
New model class for SKLearn models (training, evaluation, inference)

### PR DESCRIPTION
**Pull request by dpalencia**

### CODE REVIEW

**Comment by michaelmunje** [comment]
After this is merged, we need to make sure documentation works properly. If not, we will have to patch it by reviewing the .gitignore changes.

File: marsvision/pipeline/Model.py [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+        if "num_epochs" in kwargs:
+            self.num_epochs = kwargs["num_epochs"]
+
+        self.model_type = model_type
+        self.model = model
+        
+        # Initialize extracted features to none; use this member when we use the sklearn model
+        self.extracted_features = None
+
+    def predict(self, image: np.ndarray):
```
Comment (michaelmunje) [inline marsvision/pipeline/Model.py:46]: This function should have an input of an array of images.

File: marsvision/pipeline/SlidingWindow.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        self.create_sql_table()
+        self.write_global_to_sql(filename)
+
+        #  Get the primary key (auto incremented integer) of the new table we just wrote
+        #  So that we can pass it onto the window
+        c = self.conn.cursor()
+        c.execute("SELECT id FROM global ORDER BY id DESC LIMIT 1")
+        global_id = c.fetchone()[0]
+
+        for y in range(0, image.shape[0], self.stride_x):
```
Comment (michaelmunje) [inline marsvision/pipeline/SlidingWindow.py:67]: Rather than a loop we are going to want to send a batch of images to this inference function.

File: marsvision/pipeline/SlidingWindow.py [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+            for x in range(0, image.shape[1], self.stride_y):
```
Comment (michaelmunje) [inline marsvision/pipeline/SlidingWindow.py:68]: If the image height cannot fit N stride_y's exactly, then some portion of the image will never be reached. Therefore, there is an edge case for the very end regions of the image (rightmost or bottommost).

**Review by michaelmunje — CHANGES_REQUESTED** [review]
Added a few requests but other than that everything looks good and should be ready to merge soon.

File: marsvision/pipeline/Model.py (hunk 1 above)
Comment (dpalencia) [reply marsvision/pipeline/Model.py]: Recatored to handle an array of images.

**Comment by dpalencia** [comment]
Documenting this for later--I believe the testing suite assertion is failing on the last commit because of discrepancies in SKLearn versions.

>   /home/travis/miniconda/envs/marsvision/lib/python3.7/site-packages/sklearn/base.py:315: UserWarning: Trying to unpickle estimator LogisticRegression from version 0.23.2 when using version 0.24.1. This might lead to breaking code or invalid results. Use at your own risk.

File: marsvision/pipeline/SlidingWindow.py (hunk 3 above)
Comment (dpalencia) [reply marsvision/pipeline/SlidingWindow.py]: These should be handled by lines 82-83, where the windows are taken to be min(x + window_width, image_width) for the horizontal axis; same thing for the vertical axis. This behavior can be seen in the sliding window notebook, where the windows on the edges are thin slices.

**Review by michaelmunje — APPROVED** [review]
